crates/state/src/global_state.rs

```diff
@@ -1,20 +1,24 @@
+use tokio::runtime::Runtime;
+
+use std::sync::{Arc, Mutex, MutexGuard};
+
 use svm_codec::Codec;
 use svm_hash::{Blake3Hasher, Hasher};
+use svm_types::{Layer, State};
 
-use svm_layout::{FixedLayout, Id};
-use svm_types::{Address, BytesPrimitive, Layer, Sections, TemplateAddr};
-
-use crate::account_data::{AccountData, AccountMut};
-use crate::storage::{Fingerprint, Storage};
+use crate::storage::Storage;
 use crate::{StorageError, StorageResult as Result};
 
 /// A key-value store with a non-falsifiable state signature, historical data
 /// querying and other features which make it suitable for storing Spacemesh'
 /// global state.
 ///
 /// This data structure is backed by SQLite.
+#[derive(Debug, Clone)]
 pub struct GlobalState {
-    pub(crate) storage: Storage,
+    pub(crate) storage: Arc<Mutex<Storage>>,
+    runtime: Arc<Mutex<Runtime>>,
+    historical_query_context: Option<Layer>,
```

Comment:
this name feels a bit too wordy - is it like the `HEAD` reference used in Git? (metaphorically)

Reply:
Yes. I agree it's way too verbose, but this is complex logic and I opted for the more descriptive name. Any tips?

Reply:
let's open an issue for that in order to avoid blocking this PR

Reply:
https://github.com/spacemeshos/svm/issues/368